tower-service/src/lib.rs

```diff
@@ -348,36 +348,22 @@ pub trait Service<Request> {
     fn call(&mut self, req: Request) -> Self::Future;
 }
 
-impl<'a, S, Request> Service<Request> for &'a mut S
+impl<Request, F, Fut, T, E> Service<Request> for F
 where
-    S: Service<Request> + 'a,
+    F: FnMut(Request) -> Fut,
+    Fut: Future<Output = Result<T, E>>,
 {
-    type Response = S::Response;
-    type Error = S::Error;
-    type Future = S::Future;
+    type Response = T;
+    type Error = E;
+    type Future = Fut;
 
-    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), S::Error>> {
-        (**self).poll_ready(cx)
+    #[inline]
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
```

Comment:
My concern is that this would be really easy to forget to forward the `poll_ready`, no?

Reply:
what do you mean exactly? a service implemented for a `Fn` cannot, by definition, have `poll_ready`; if it ever called into another service, it would have to call `ServiceExt::ready` or something first. I think this is definitely something we would want to document if we moved forward with this, but the same issue existed previously with `ServiceFn`-type adapters...

Reply:
Yeah, you're right it does currently exist in the current iteration. I guess the problem I am worried about is more so for layers. So nvm!